packages/fiori-annotation-api/src/cds/deletion.ts

```diff
@@ -258,7 +258,7 @@ function mergeDeletionRanges(tokens: CompilerToken[], deletionRanges: DeletionRa
         do {
             doMerge = false;
             const next = rangeIndex + 1 < deletionRanges.length ? deletionRanges[rangeIndex + 1] : null;
-            if (next && current.termRange.end + 1 === next.termRange.start) {
+            if (current.termRange.end + 1 === next?.termRange.start) {
```

Comment:
`?` needs to be before all segments otherwise it will throw if `next` is undefined.
```suggestion
            if (current.termRange.end + 1 === next?.termRange?.start) {
```